### ISSUE TITLE
Updates fail under strict security

### ISSUE THREAD

**Issue by end-user** [microsoft/azuredatastudio#11793]
Issue Type: <b>Performance Issue</b>

I cannot run the auto-update as it's trying to execute under a temp directory and is being blocked by company policy.  I need a way to change the configuration, prompt to overide/elevate, or some other alternative.

Azure Data Studio version: azuredatastudio 1.20.1 (fe6312306647441b30102a64bb43abb06ebcf2be, 2020-07-17T02:39:41.016Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.81GB (5.20GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	    78	  4876	azuredatastudio main
    0	   195	  1532	   window (Settings - Azure Data Studio)
    0	    78	 16552	     extensionHost
    0	    31	 14148	       "c:\Users\oblioleitch\AppData\Local\Programs\Azure Data Studio\resources\app\extensions\mssql\sqltoolsservice\Windows\3.0.0-release.4\MicrosoftSqlToolsCredentials.exe" --log-file c:\Users\oblioleitch\AppData\Roaming\azuredatastudio\logs\20200813T103844\exthost1\Microsoft.mssql\credentialstore.log --tracing-level Critical
    0	    11	  2264	         console-window-host (Windows internal process)
    0	    34	 22084	       "c:\Users\oblioleitch\AppData\Local\Programs\Azure Data Studio\resources\app\extensions\mssql\sqltoolsservice\Windows\3.0.0-release.4\SqlToolsResourceProviderService.exe" --log-file c:\Users\oblioleitch\AppData\Roaming\azuredatastudio\logs\20200813T103844\exthost1\Microsoft.mssql\resourceprovider.log --tracing-level Critical
    0	    11	 26836	         console-window-host (Windows internal process)
    0	    61	 24108	       "c:\Users\oblioleitch\AppData\Local\Programs\Azure Data Studio\resources\app\extensions\mssql\sqltoolsservice\Windows\3.0.0-release.4\MicrosoftSqlToolsServiceLayer.exe" --log-file c:\Users\oblioleitch\AppData\Roaming\azuredatastudio\logs\20200813T103844\exthost1\Microsoft.mssql\sqltools.log --tracing-level Critical
    0	    11	 11856	         console-window-host (Windows internal process)
    0	    71	 14048	   shared-process
    0	   162	 14828	   gpu-process
    1	    74	 16612	   window (Issue Reporter)
    0	    25	 26512	   utility
```

</details>
<details>
<summary>Workspace Info</summary>

```
;
```

</details>
<details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by Jazzynupe** [comment]
Also seeing the same issue on my laptop as well when trying to run the update through Azure Data Studio or manually through download. Would need a command line switch to change the temp directory for running this update.

**Comment by end-user** [comment]
further explanation:
![image](https://user-images.githubusercontent.com/4923193/92126453-32d3ab00-edce-11ea-81fd-3d9b7440d3c5.png)
which is caused by:
![image](https://user-images.githubusercontent.com/4923193/92126590-5dbdff00-edce-11ea-9dc4-4a7b514cb8f6.png)

**Comment by Jazzynupe** [comment]
Just tried update 1.21.0 (Date: 2020-08-11) and it installed successfully.
If you have not tried it recently, you may want to try it again.

This is using VS Code 1.48.0

**Comment by kburtram** [comment]
@end-user are you still seeing this on more recent builds?

@abist any ideas what we can do investigate this?

**Comment by abist** [comment]
@kburtram Let me investigate if there were any changes in the setup process upstream

**Comment by abist** [comment]
Closing this since it was fixed in 1.21.0.
